ssl/ssl_session.cc

```diff
@@ -661,11 +661,15 @@ static enum ssl_hs_wait_t ssl_lookup_session(
           MakeConstSpan(sess->session_id, sess->session_id_length);
       return key_id == sess_id ? 0 : 1;
     };
-    MutexReadLock lock(&ssl->session_ctx->lock);
+    MutexWriteLock lock(&ssl->session_ctx->lock);
```

Comment:
This is locking the SSL_CTX's session context lock, should this stay the same and we let the ssl_update_counter get the SSL_CTX lock in that funciton?

Reply:
The function is going to be using the same `SSL_CTX` lock, since the stats are within the same `ssl->session_ctx`. If we let `ssl_update_counter` update the lock, the tests processes will crash (which is expected since we're trying to lock the mutex twice).